### PR TITLE
Added TDFN 10 Package with EP

### PR DESCRIPTION
**Pull request by jamesrlucas**
I've added the definitions for a TDFN 10 package with an exposed pad.

Package Drawing: https://pdfserv.maximintegrated.com/package_dwgs/21-0429.PDF

[Footprint PR](https://github.com/KiCad/kicad-footprints/pull/2248)
[Symbol PR](https://github.com/KiCad/kicad-symbols/pull/2360)

image

image

### CODE REVIEW

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 1]
```diff
@@ -687,3 +687,53 @@ TDFN-6-1EP_2.5x2.5mm_P0.65mm_EP1.3x2mm:
...
+    tolerance: 0.1
+
+  EP_size_x:
+    nominal: 0.9
+    tolerance: 0.1
+  EP_size_y:
+    nominal: 2.0
+    tolerance: 0.1
+
+  EP_num_paste_pads: [2, 2]
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:722]: I suggest a single paste pad in x direction.
Currently, there's 90µm clearance between them, probably not doing much good but requiring a more delicate (and expensive) stencil.

File: scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml [hunk 2]
```diff
@@ -687,3 +687,53 @@ TDFN-6-1EP_2.5x2.5mm_P0.65mm_EP1.3x2mm:
...
+
+  thermal_vias:
+    count: [1, 3]
+    drill: 0.2
```
Comment (chschlue) [inline scripts/Packages/Package_NoLead__DFN_QFN_LGA_SON/size_definitions/dfn.yaml:726]: How about two 0.3mm vias in y direction instead?

**Comment by jamesrlucas** [comment]
I'll sort the paste now.

In the component [datasheet](https://datasheets.maximintegrated.com/en/ds/MAX17501.pdf) it shows three vias (page 17, example layout), so just went with that.

**Comment by chschlue** [comment]
FP merged.

**Comment by evanshultz** [comment]
@jamesrlucas 
If you can fix the branch conflict I will merge this.

**Comment by jamesrlucas** [comment]
Okay, fixed it :) 

James

**Comment by evanshultz** [comment]
Thank you!

**Pull request merged**